internal/moduletest/eval_context.go

```diff
@@ -198,6 +204,23 @@ func (ec *EvalContext) Evaluate() (Status, cty.Value, tfdiags.Diagnostics) {
 	return status, cty.ObjectVal(outputVals), diags
 }
 
+func diagsForEphemeralResources(refs []*addrs.Reference) (diags tfdiags.Diagnostics) {
+	for _, ref := range refs {
+		switch v := ref.Subject.(type) {
+		case addrs.ResourceInstance:
+			if v.Resource.Mode == addrs.EphemeralResourceMode {
+				diags = diags.Append(&hcl.Diagnostic{
+					Severity: hcl.DiagError,
+					Summary:  "Ephemeral resources not supported in the context of tests",
+					Detail:   "Ephemeral resources are not supported in the context of terraform test.",
```

Comment:
I think this is a little misleading, as they are supported (as in you can use them in the configuration being tested) but they're not available to be validated.

Maybe something like `Ephemeral resources cannot be asserted` and `Ephemeral resources are closed when the test is finished, and are not available within the test context for assertions`?

Reply:
Yeah, I can see how that might be misleading